### ISSUE TITLE
Update `TimePicker` component

### ISSUE THREAD

**Issue by joshestein** [bluedotimpact/bluedot#3016]
### 🧩 Summary

Figma: https://www.figma.com/design/b48ano1xNhjFlspfahUkGn/BlueDot-Design-System?node-id=1603-2&m=dev

### ✅ Definition of done

1. Use new design tokens instead of TW defaults
2. Smaller size
4. Drop black focus ring
5. Disabled state
